src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java

```diff
@@ -133,6 +148,30 @@ public void setRegion(String region) {
         save();
     }
 
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    @DataBoundSetter
+    public void setCredentialsId(String credentialsId) {
+        this.credentialsId = credentialsId;
+        save();
+    }
+
+    public AWSCredentialsImpl getCredentials() {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        Optional<AmazonWebServicesCredentials> credential = CredentialsProvider.lookupCredentials(
+                AmazonWebServicesCredentials.class, Jenkins.get(),ACL.SYSTEM, Collections.emptyList())
+                                                                               .stream()
+                                                                               .filter(it -> it.getId().equals(credentialsId))
```

Comment:
Are you sure there is not an overload where you are passing in the `credentialsId` as a parameter? And the `Class<? extends Credentials>`? Maybe need some API usability work in `credentials`…

Reply:
I did not found an easy way to grab a credential by ID outside of a job context, but yep, probably credentials API needs to provide a simpler way to make the basics searches, we are repeating the same piece of code on all places to create the listbox and to retrieve the credential, we always want the same all the credentials that I can see of a type on all domains, or a credential by ID.